lib/src/charts/painters/cartesian/bar_painter.dart

```diff
@@ -24,17 +23,53 @@ class BarPainter implements CartesianPainter {
   late Paint _barPaint;
   late Paint _barStroke;
 
-  BarPainter({
-    required this.useGraphUnits,
-  }) {
+  BarPainter({required this.useGraphUnits}) {
     _barPaint = Paint();
     _barStroke = Paint()..style = PaintingStyle.stroke;
   }
 
+  @override
+  BarInteractionResult? hitTest(
+    TouchInteractionType type,
+    Offset localPosition,
+  ) {
+    // We will perform HitTest only if Interactions are enabled for this series.
+    if (_data.interactionEvents.isEnabled) {
+      var snapToNearestBar = _data.interactionEvents.snapToNearestPoint;
+      var snappingRange = _data.interactionEvents.snappingRange;
+
+      var foundBar = _interactionData.withNullableItems().singleWhere((data) {
+        // Check if we have to find the nearest bar to interaction position
+        if (snapToNearestBar) {
+          return data!.findNearest(localPosition, snappingRange);
+        } else {
+          return data!.contains(localPosition);
+        }
```

Comment:
As we discussed, this needs to follow these specific steps
1. Check `rect.contains` first. This will be the quickest exit path. We also need a fuzziness value. We cannot call it `snappingRange` because it is not related to snapping. If fuzziness is provided we check contains with fuzziness.
2. If contains is false and snapping is true then try to find the nearest bar by first checking between which 2 bars does the offset lie and then calculating the closest distance to find that bar